SDK / MM Connect new structure (#2337)
MetaMask/metamask-docs: sdk/index.md

MetaMask Connect fills that gap and gives you more:

1. **Cross-platform connections** — Reach users on any device. When the extension isn't available, the SDK connects to MetaMask Mobile through a relay, with no extra setup on your side.
2. **Multichain sessions** — Request access to EVM and Solana (and future ecosystems) in a single connection, instead of connecting per chain.
3. **Persistent sessions** — Sessions survive page reloads and new tabs, so users aren't prompted to reconnect.
4. **Consistent API** — The same interface works whether the user connects through the extension or Mobile.

## Get started

<CardList
items={[
{
href: '/sdk/multichain',
title: 'Multichain',
description: 'Connect to multiple ecosystems at the same time.',
},
{
href: '/sdk/evm',
title: 'EVM networks',
description: 'Connect to Ethereum and other EVM networks.',
},
{
href: '/sdk/solana',
title: 'Solana',
description: 'Connect to Solana.',
}
]}
/>

## Integration options

There are three ways to integrate, depending on what works best for your dapp:

### Option A: Single-ecosystem client (drop-in for existing dapps)

> If your dapp already targets a single ecosystem, use [`@metamask/connect-evm`](/sdk/evm) or [`@metamask/connect-solana`](/sdk/solana) for a familiar experience with minimal changes to your existing code.

Reviewer: ```suggestion
If your dapp targets a single ecosystem, use [`@metamask/connect-evm`](/sdk/evm) or [`@metamask/connect-solana`](/sdk/solana) for a familiar experience with minimal changes to your existing code.
```

- **EVM** — Gives you an [EIP-1193](https://eips.ethereum.org/EIPS/eip-1193) compatible provider, so your existing `ethers.js`, `viem`, or `web3.js` code works as-is.

> - **Solana** — Gives you a [Wallet Standard](https://github.com/wallet-standard/wallet-standard) compatible wallet that plugs into the Solana wallet adapter ecosystem.

Reviewer: ```suggestion
- **Solana**: Provides a [Wallet Standard](https://github.com/wallet-standard/wallet-standard) compatible wallet that integrates with the Solana wallet adapter ecosystem.
```

This is the fastest path if you already have a working single-chain dapp and want to add MetaMask Connect.

### Option B: Multiple ecosystem clients

> If your dapp needs to support both EVM and Solana, you can use [`@metamask/connect-evm`](/sdk/evm) and [`@metamask/connect-solana`](/sdk/solana) together.

Reviewer: ```suggestion
If your dapp supports both EVM and Solana, use [`@metamask/connect-evm`](/sdk/evm) and [`@metamask/connect-solana`](/sdk/solana) together.
```

This gives you per-ecosystem APIs while covering multiple chains, and is a good fit when you want to keep familiar provider interfaces for each ecosystem.

### Option C: Multichain client (full control, recommended)

> Use [`@metamask/connect-multichain`](/sdk/multichain) directly for the full [Multichain API](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-25.md).

Reviewer: ```suggestion
Use [`@metamask/connect-multichain`](/sdk/multichain) to work directly with the [Multichain API](https://github.com/ChainAgnostic/CAIPs/blob/main/CAIPs/caip-25.md).
```

This gives you more control — you work with **scopes** and `wallet_invokeMethod` rather than per-chain RPC — and unlocks the best experience for multichain dapps with a single connection prompt for all ecosystems.
**This is the recommended long-term path.**

All three options are built on the same multichain client under the hood, sharing the same transport and session infrastructure.

> You can start with whichever option fits your dapp today and migrate to **Option C** when you're ready for full multichain control.

Reviewer: ```suggestion
All options use the same underlying multichain client and share transport and session infrastructure.
You can start with the option that fits your dapp today and move to the multichain client (Option C) when you need full multichain control.
```
